src/Common/src/Microsoft/Win32/SafeHandles/SafeRsaHandle.Unix.cs

```diff
@@ -26,5 +27,19 @@ public override bool IsInvalid
         {
             get { return handle == IntPtr.Zero; }
         }
+
+        internal static SafeRsaHandle DuplicateHandle(IntPtr handle)
+        {
+            Debug.Assert(handle != IntPtr.Zero);
+
+            if (!Interop.libcrypto.RSA_up_ref(handle))
+            {
+                throw Interop.libcrypto.CreateOpenSslCryptographicException();
+            }
+
+            SafeRsaHandle safeHandle = new SafeRsaHandle();
```

Comment:
Small reliability suggestion: move this allocation to before the RSA_up_ref call.  That way, if we're running low on memory and the allocation were to throw, it'll happen before we increase the ref count and we won't leak the handle.